src/time_evolution/time_evolution.jl

```diff
@@ -95,6 +96,37 @@ function Base.show(io::IO, sol::TimeEvolutionMCSol)
     return nothing
 end
 
+struct TimeEvolutionSSESol{
+    TT<:Vector{<:Real},
+    TS<:AbstractVector,
+    TE<:Matrix{ComplexF64},
+    TEA<:Array{ComplexF64,3},
+    T<:Real,
+}
+    n_traj::Int
+    times::TT
+    states::TS
+    expect::TE
+    expect_all::TEA
+    converged::Bool
+    alg::StochasticDiffEq.StochasticDiffEqAlgorithm
+    abstol::T
+    reltol::T
+end
```

Comment:
I think we don't need to specify `T<:Real`, because now it restrict the type of `abstol` and `reltol` to be exactly the same.
How about doing it in this way:
```julia
abstol::Real
reltol::Real
```
like the way we did in [`TimeEvolutionMCSol`](https://github.com/qutip/QuantumToolbox.jl/blob/main/src/time_evolution/time_evolution.jl#L64)

Reply:
We thought that the last option would lead to type instabilities. Do you agree?

Reply:
If we remove `T<:Real`, I don't think it will cause type instabilities.
Like we did in [TimeEvolutionMCSol](https://github.com/qutip/QuantumToolbox.jl/blob/main/src/time_evolution/time_evolution.jl#L64)

Reply:
So the whole thing becomes
```julia
struct TimeEvolutionSSESol{
    TT<:Vector{<:Real},
    TS<:AbstractVector,
    TE<:Matrix{ComplexF64},
    TEA<:Array{ComplexF64,3}
}
    n_traj::Int
    times::TT
    states::TS
    expect::TE
    expect_all::TEA
    converged::Bool
    alg::StochasticDiffEq.StochasticDiffEqAlgorithm
    abstol::Real
    reltol::Real
end
```

Reply:
Yeah, I agree, indeed I think we should also change it to the other strucs. Then how the compiler knows about the type of abstol and reltol?

See for example this simple code
```julia
a = destroy(10)
sol = sesolve(a' * a, fock(10, 0), range(0, 1, 1000))
@code_warntype sol.abstol
```

```julia
MethodInstance for getproperty(::TimeEvolutionSol{Vector{Float64}, Vector{QuantumObject{Vector{ComplexF64}, KetQuantumObject}}, Matrix{ComplexF64}}, ::Symbol)
  from getproperty(x, f::Symbol) @ Base Base.jl:37
Arguments
  #self#::Core.Const(getproperty)
  x::TimeEvolutionSol{Vector{Float64}, Vector{QuantumObject{Vector{ComplexF64}, KetQuantumObject}}, Matrix{ComplexF64}}
  f::Symbol
Body::Any
1 ─      nothing
│   %2 = Base.getfield(x, f)::Any
└──      return %2
```

For the moment we defined two different types, `T1` and `T2`.